Dev

merrilmathew opened the pull request:
This PR includes github actions to deploy monitoring solution to prod environment. Added variable files and updated Makefile to include prod target.

Commit:
Add terraform support to Makefile

The Makefile should have targets which supports terraform commands for
different environments. This commit adds targets so it can be consumed
locally and by Github actions.

chelnak commented on terraform/workspace_variables/preprod.backend.tfvars:
```diff
@@ -0,0 +1,3 @@
+storage_account_name = "s165t01preprodtfstate"
+key                  = "preprod_monitoring.tfstate"
+resource_group_name  = "s165t01-preprod-rg"
```
Missing a new line

chelnak commented on terraform/workspace_variables/preprod.tfvars.json:
```diff
@@ -0,0 +1,13 @@
+    "monitor_scrape_backing_services": [
+      "tra-test/qualified-teachers-api-preprod-pg-svc"
+    ]
+}
```
Missing a new line

chelnak commented on terraform/workspace_variables/prod.backend.tfvars:
```diff
@@ -0,0 +1,3 @@
+storage_account_name = "s165p01tfstate"
+key                  = "prod_monitoring.tfstate"
+resource_group_name  = "s165p01-rg"
```
Missing a new line

chelnak commented on terraform/workspace_variables/prod.tfvars.json:
```diff
@@ -0,0 +1,13 @@
+    "monitor_scrape_backing_services": [
+      "tra-production/qualified-teachers-api-prod-pg-svc"
+    ]
+}
```
Missing a new line

chelnak commented on terraform/workspace_variables/test.tfvars.json:
```diff
@@ -0,0 +1,13 @@
+    "monitor_scrape_backing_services": [
+      "tra-test/qualified-teachers-api-test-pg-svc"
+    ]
+  }
```
Missing a new line

chelnak commented on .github/workflows/build.yml:
```diff
@@ -28,9 +29,94 @@ jobs:
         terraform validate -no-color
       working-directory: terraform
-
+  
```
is this whitespace char needed?

saliceti replied:
@merrilmathew you may want to enable this option in VScode:
- Trim trailing whitespace

chelnak commented on .github/workflows/build.yml:
```diff
@@ -28,9 +29,94 @@ jobs:
+    needs: deploy_preprod
+    runs-on: ubuntu-latest
+    environment: prod
+    # if: github.ref == 'refs/heads/main'
```
Is this redundant now?

merrilmathew replied:
Removed. There is environment check instead

chelnak commented on .github/workflows/build.yml:
```diff
@@ -28,9 +29,94 @@ jobs:
+      uses: ./.github/workflows/actions
+      with:
+        environment_name: prod
+        azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
```
Missing a new line

saliceti replied:
@merrilmathew you may want to enable these options in VScode:
- Files: Insert final newline
- Files: Trim final newlines

chelnak reviewed (changes requested):
Some minor changes requested & have asked for clearer commit messages

saliceti commented on .github/workflows/build.yml:
```diff
@@ -28,9 +29,94 @@ jobs:
+      with:
+        terraform_version: 1.0.10
+
+    - name: Check formatting
```
Formatting and validation were already done in dev, we probably don't need to do it again

merrilmathew commented:
Think I resolved the suggestions.

saliceti commented on .github/workflows/build.yml:
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
     - dev
+    - main
```
Can we leave `main` only?

merrilmathew replied:
removed dev

saliceti commented on .github/workflows/destroy/action.yml:
```diff
@@ -0,0 +1,41 @@
+name: Destroy PAAS environment
```
As discussed, we can remove this action and use the make commands instead if necessary

merrilmathew replied:
removed action and rules from Makefile

merrilmathew committed:
Update workflow to support prometheus deployment

Update build.yml to inlclude steps that uses the Makefile
to run terraform commands to deploy prometheus.

merrilmathew committed:
Add terraform config to deploy prometheus prod

The previous PR added dev support and this commit adds support for
deploying production resources to PAAS.

chelnak commented:
10/10 would read the commits again 👍
